aten/src/ATen/native/cuda/Nonzero.cu

```diff
@@ -18,19 +18,60 @@
 
 namespace at::native {
 
-namespace{
+namespace {
+// Wrapper for DeviceSelect::If to handle tensors larger than INT_MAX
+// Imported from https://github.com/NVIDIA/cccl/pull/1379
+// #TODO: Remove the wrapper when https://github.com/NVIDIA/cccl/issues/1422 is released
+template <typename InputIteratorT,
+          typename OutputIteratorT,
+          typename NumSelectedIteratorT,
+          typename OffsetT,
+          typename SelectOp>
+static cudaError_t dispatch_select_if_wrapper(
+  void* d_temp_storage,
+  std::size_t& temp_storage_bytes,
+  InputIteratorT d_in,
+  OutputIteratorT d_out,
+  NumSelectedIteratorT d_num_selected_out,
+  OffsetT num_items,
+  SelectOp select_op,
+  cudaStream_t stream = 0)
+{
+    using flag_iterator_t = cub::NullType*;
+    using equality_op_t   = cub::NullType;
+
+    return cub::DispatchSelectIf<
```

Comment:
Do you we need to wait for https://github.com/pytorch/pytorch/pull/125997 for a compatible API?

Reply:
Or cub/cccl 2.4.0?